src/Analysis/Ast/Impl/Types/PythonClassType.cs

```diff
@@ -192,12 +191,15 @@ internal PythonClassType(string name, Location location)
 
         #endregion
 
-        internal void SetBases(IEnumerable<IPythonType> bases) {
+        internal void SetBases(IEnumerable<IPythonType> bases, IExpressionEvaluator eval = null) {
             if (_bases != null) {
                 return; // Already set
             }
 
-            bases = bases != null ? bases.Where(b => !b.GetPythonType().IsUnknown()).ToArray() : Array.Empty<IPythonType>();
+            // Consider
+            //    from X import A
+            //    class A(A): ...
+            bases = DisambiguateBases(bases, eval).ToArray();
```

Comment:
perhaps just a name change to `ResolveBases` or maybe even consider just putting these methods in `ClassEvaluator` because much of the work that it is doing is similar to the `ClassEvaluator.ProcessBases`

Reply:
`SetBases` is internal so every call should guarantee that there won't be cycles. `ProcessBases` is just happens to be one of the cases. Otherwise every caller to `SetBases` will have to remember to resolve.

Reply:
if eval is being passed into set bases then the work done in process bases can probably be put into set bases

Reply:
I doesn't actually need eval, only scope.

Reply:
if eval is there then I think it would be better to move all processing inside SetBases. Otherwise we end up with the case where sometimes the bases are fully processed and sometimes they aren't

Reply:
Replaced by scope